hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -194,21 +194,37 @@ public Record getRecord(Data key) {
     public Record putOrUpdateReplicatedRecord(Data dataKey, Record replicatedRecord,
                                               ExpiryMetadata expiryMetadata,
                                               boolean indexesMustBePopulated, long now) {
-        Record newRecord = storage.get(dataKey);
-        if (newRecord == null) {
-            newRecord = createRecord(dataKey, replicatedRecord != null
+        // Ensure the replicatedRecord is kept in-memory until being released
+        storage.pinRecord(replicatedRecord);
```

Comment:
🤔  from what I understand, this method will be called only during map replication (e.g., migration) on the target, where `replicatedRecord` is just a standalone memoryBlock/record, so I was wondering why do we need pinning of it?

Reply:
You're right, simplified in https://github.com/hazelcast/hazelcast/pull/20274/commits/fc08d0f63f52ec102c5797f3f95b6bf2568a5d00